viz/MLSMapVisualization.cpp

```diff
@@ -269,11 +444,39 @@ void MLSMapVisualization::updateMainNode ( osg::Node* node )
     // draw the extents of the mls
     visualizeMapExtents(p->getCellExtents(), p->getResolution());
 
+    geode->setUseVerticalTopColor(use_vertical_top_color);
+
     if(cycleHeightColor)
     {
+        if (auto_color_cycle_interval) {
+            cycleColorInterval = p->getMax() - p->getMin();
+            cycleColorIntervalUniform->set((float)cycleColorInterval);
+        }
         geode->showCycleColor(true);
         geode->setCycleColorInterval(cycleColorInterval);
         geode->setColorHSVA(0, 1.0, 0.6, 1.0);
+
+        if (use_shader_color) {
+            // enable shader-based height coloring
+            geode->getOrCreateStateSet()->setAttributeAndModes(program.get(), osg::StateAttribute::ON);
+
+            osg::ref_ptr<osg::Uniform> mvp = new osg::Uniform(osg::Uniform::FLOAT_MAT4, "modelViewProjectionMatrix");
+            geode->getOrCreateStateSet()->addUniform(mvp);
+            osg::Camera* cam = getCamera();
```

Comment:
Add this line the viz-unit tests fail (it might be, because the standalone visualizer does not behave as expected)

Reply:
getCamera() is currently only availabe in the qt4-qt5 branch of vizkit3d 

https://github.com/rock-core/gui-vizkit3d/commit/495a2a33c56df53a3a66bf3475773b9acf673733

Reply:
It was a runtime crash on Ubuntu 24 (I am on the qt4-qt5 branch of vizkit3d of course, but I did not recompile that with Debug-Info, so I'm not sure where exactly it failed).
Can you try building (and running) the unit tests? (The GUI unit-tests require to manually close the window, and some seem not to clear the previous visualization -- but ideally they should not crash ^^)

Reply:
Found the issue: The test case uses vizkit3d::QtThreadedWidget within StandaloneVisualizer.cpp.
When the plugin is loaded this way, this messsage is printed:

> QObject::setParent: Cannot set parent, new parent is in a different thread when trying to set the parent widget 

But the shader needs to get the osg::camera from the parent widget of the plugin.

I just fixed the crash by disabling the shaders in case the camera cannot be ontained.